packages/sync/src/manager.ts

```diff
@@ -258,30 +346,39 @@ export function createSyncManager(): SyncManager {
 	 */
 	function updateCRDTDoc(
 		objectType: ObjectType,
-		objectId: ObjectID,
+		objectId: ObjectID | null,
 		changes: Partial< ObjectData >,
 		origin: string,
 		isSave: boolean = false
 	): void {
 		const entityId = getEntityId( objectType, objectId );
 		const entityState = entityStates.get( entityId );
+		const collectionState = collectionStates.get( objectType );
 
-		if ( ! entityState ) {
-			return;
-		}
-
-		const { syncConfig, ydoc } = entityState;
+		if ( entityState ) {
+			const { syncConfig, ydoc } = entityState;
+			ydoc.transact( () => {
+				syncConfig.applyChangesToCRDTDoc( ydoc, changes );
 
-		ydoc.transact( () => {
-			syncConfig.applyChangesToCRDTDoc( ydoc, changes );
+				if ( isSave ) {
+					// Mark the document as saved in the record metadata map.
+					const recordMeta = ydoc.getMap( RECORD_METADATA_KEY );
```

Comment:
Same thing here for reducing the duplication here. Could move the isSave block to a separate method and call that here, as well as under collectionState as it's the same thing.